test/Garnet.test/Extensions/RateLimiterTxn.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Text;
+using Garnet.common;
+using Garnet.server;
+using Tsavorite.core;
+
+namespace Garnet
+{
+    sealed class RateLimiterTxn : CustomTransactionProcedure
+    {
+        public override bool Prepare<TGarnetReadApi>(TGarnetReadApi api, ref CustomProcedureInput procInput)
+        {
+            int offset = 0;
+            AddKey(GetNextArg(ref procInput, ref offset), LockType.Exclusive, true);
```

Comment:
Wanted to understand if there is a way to have configurable server side timeouts. There might be a scenario with 1000s of requests for same key waiting to obtain lock. In such cases, want to have timeouts so that load can be reduced on server.

Reply:
You can override this field from CustomTransactionProcedure.cs base class --

```cs
        /// <summary>
        /// Timeout for acquiring key locks. 100 ms by default
        /// </summary>
        public virtual TimeSpan KeyLockTimeout => TimeSpan.FromMilliseconds(
#if DEBUG
            5000    /* Things are slower in Debug so give it more time */
#else
            100
#endif
        );
```